src/App.tsx

```diff
@@ -1,73 +1,111 @@
-import React from 'react';
+import React, { useMemo, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import debounce from 'lodash.debounce';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [inputValue, setInputValue] = useState('');
+  const [suggestions, setSuggestions] = useState<Person[]>([]);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [isDropdownVisible, setIsDropdownVisible] = useState(false);
+
+  const debouncedSuggestions = useMemo(
+    () =>
+      debounce((value: string) => {
+        const filteredSuggestions = value
+          ? peopleFromServer.filter(person => person.name.includes(value))
+          : peopleFromServer;
+
+        setSuggestions(filteredSuggestions);
+      }, 300),
+    [],
+  );
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputValue(event.target.value);
+    setSelectedPerson(null);
+    debouncedSuggestions(event.target.value);
+    setIsDropdownVisible(true);
+  };
+
+  const handleSuggestionClick = (person: Person) => {
+    setInputValue(person.name);
+    setSelectedPerson(person);
+    setSuggestions([]);
+    setIsDropdownVisible(false);
+  };
+
+  const handleInputFocus = () => {
+    setIsDropdownVisible(true);
+    if (!inputValue) {
+      setSuggestions(peopleFromServer);
+    }
+  };
 
   return (
     <div className="container">
       <main className="section is-flex is-flex-direction-column">
-        <h1 className="title" data-cy="title">
-          {`${name} (${born} - ${died})`}
-        </h1>
+        {selectedPerson ? (
+          <h1 className="title" data-cy="title">
+            {`${selectedPerson.name} (${selectedPerson.born} - ${selectedPerson.died})`}
+          </h1>
+        ) : (
+          <h1 className="title" data-cy="title">
+            No selected person
+          </h1>
+        )}
```

Comment:
You don't need to duplicate this tag twice, you need to put the check inside the tag. It would also be better to save this check in a separate variable and use it here
```suggestion
        {selectedPerson ? (
          <h1 className="title" data-cy="title">
            {`${selectedPerson.name} (${selectedPerson.born} - ${selectedPerson.died})`}
          </h1>
        ) : (
          <h1 className="title" data-cy="title">
            No selected person
          </h1>
        )}
```